src/v/datalake/datalake_manager.cc

```diff
@@ -260,14 +256,17 @@ void datalake_manager::start_translator(
     _translators.emplace(partition->ntp(), std::move(translator));
 }
 
-ss::future<> datalake_manager::stop_translator(const model::ntp& ntp) {
+void datalake_manager::stop_translator(const model::ntp& ntp) {
     auto it = _translators.find(ntp);
     if (it == _translators.end()) {
-        co_return;
+        return;
     }
-    auto translator = std::move(it->second);
+    auto t = std::move(it->second);
     _translators.erase(it);
-    co_await translator->stop();
+    ssx::spawn_with_gate(_gate, [t = std::move(t)]() mutable {
+        auto* t_ptr = t.get();
+        return t_ptr->stop().finally([_ = std::move(t)] {});
```

Comment:
what happens to the partition translator that is running if `_gate` is closed?

Reply:
Any translator still in the map will be stopped in `datalake_manager::stop()`. Because we do the map update synchronously now in `stop_translator`, we should be guaranteed that the translator is stopped here or in the class-level stop

Reply:
Oh actually I think that's not true still because `stop_translator` doesn't take the gate

Reply:
> we should be guaranteed that the translator is stopped here or in the class-level stop

presumably the reference to it were just removed a few lines above

>    _translators.erase(it);

Reply:
yeh if you wrap more in the gate it would probably work ok

Reply:
Yea, i'll add an early return (seems cleaner than just a gate hold + exception? at least, it's one fewer exception to worry about)

Reply:
.. or early return in stop_translator() if gate is closed.

Reply:
Done

Also added a note in the commit message of the new commit: i don't think this is problematic in practice, but still it seems easy enough to add anyway

Reply:
>i don't think this is problematic in practice

I take it back, I bet there's a way for the `_iceberg_commit_interval.watch()` lambda to get unlucky and hit this since it spawns calls to `on_group_notification` in the background. It may may prevent the gate from being destructed with its hold, but it seems possible for its lambda to still be running after a call to `stop()` has been started. All the more reason to leave it in